### PR TITLE
[IRRC-101] abstract out suggested teacher modal

### PR DESCRIPTION
**Pull request by lauratsui**
Created new suggested teacher modal card used in Student Profile page (for unmatched student). Updates student and teacher statuses after matching.

Test Steps
- Create a new Student with Native Language Chinese
- Create a new Teacher with Native Language Chinese
- Navigate to Student Profile page
- Click on button to Match (Modal appears)
- Select Teacher and Match (Notification appears)

Works on any Native Language, just make sure that the Suggested Teacher Logic is satisfied, so there are options in the Teacher modal. So there must be an intersection of the Language between Student and Teacher.

### CODE REVIEW

**Comment by lauratsui** [comment]
Abstracted out suggested teacher modal that is used in student profile (status card unmatched). Notification added when matching is successful/unsuccessful.
Note that student profile page & status card unmatched still says "no teacher assigned" even after matching is successful - will handle in a separate PR in future.

File: frontend/src/pages/Teachers.vue [hunk 1]
```diff
@@ -4,44 +4,6 @@
       <a class="logo">Teachers</a>
     </PageHeader>
 
-    <section>
-      <b-table
-        :data="teachersData"
-        :selected.sync="selected"
-        @dblclick="goToTeacher"
```
Comment (therealyingtong) [inline frontend/src/pages/Teachers.vue]: This deletes the table with the `dblclick` event handler. Let's merge this PR and reintroduce the event handler on the surviving table. (Refer to #126 which was closed in favour of this PR.)

Comment (thchia) [reply frontend/src/pages/Teachers.vue]: Can we take this as an approval?

**Review by therealyingtong — APPROVED** [review]

File: frontend/src/pages/Teachers.vue (hunk 1 above)
Comment (lauratsui) [reply frontend/src/pages/Teachers.vue]: We edited this PR to reintroduce the event handler.
Also took the chance to update the style for the table (all text in same row will turn white when selected, except for status description) for readability.

**Review by dzldebby — CHANGES_REQUESTED** [review]
**OUTDATED - PLEASE IGNORE**

**Review by dzldebby — APPROVED** [review]
Please ignore the earlier review, I was on the wrong branch! Tested and works well :)

**Pull request merged**